packages/peregrine/src/hooks/useWindowSize.js

```diff
@@ -0,0 +1,31 @@
+import { useState, useEffect } from 'react';
+
+const getSize = () => {
+    return {
+        innerHeight: window.innerHeight,
+        innerWidth: window.innerWidth,
+        outerHeight: window.outerHeight,
+        outerWidth: window.outerWidth
+    };
+};
+
+/**
+ * A hook that will return inner and outer height and width values whenever
+ * the window is resized.
+ */
+export const useWindowSize = () => {
+    const [windowSize, setWindowSize] = useState(getSize());
+
+    const handleResize = () => {
+        setWindowSize(getSize());
+    };
+
+    useEffect(() => {
+        window.addEventListener('resize', handleResize);
```

Comment:
Could be an opportunity to generalize `useDocumentListener` to `useEventListener` and pass the object you want to subscribe to (window, document, element).

Reply:
That feels like overkill to me.

Reply:
I didn't realize we already had the `useDocumentListener` when I wrote that comment. I think I'll give it a shot :D